### PR TITLE
Recursive type checking

### PR DESCRIPTION
**Pull request by FWDekker**
* Adds a `getValues` function in `JimpleNode` that recursively finds _all_ `Value`s in the node.
* Changes the `generalizedValuesAreEqual` function to use `getValues` rather than `getTopLevelValues`.
* Adds tests for both functions.

### CODE REVIEW

File: modules/models/jimple-library-usage-graph/src/test/kotlin/org/cafejojo/schaapi/models/libraryusagegraph/jimple/GeneralizedNodeComparatorValueTest.kt [hunk 1]
```diff
@@ -155,5 +155,55 @@ internal class GeneralizedNodeComparatorValueTest : Spek({
...
+                val node1B = JimpleNode(mockInvokeStmt(expr1B))
+                comparator.generalizedValuesAreEqual(node1A, node1B)
+
+                val expr2A = SimpleInvokeExpr("base", method, "arg1", "arg2")
+                val expr2B = SimpleInvokeExpr("base", method, "arg1", "arg2")
+                val node2A = JimpleNode(mockInvokeStmt(expr2A))
+                val node2B = JimpleNode(mockInvokeStmt(expr2B))
+                assertThat(comparator.generalizedValuesAreEqual(node2A, node2B)).isTrue()
+            }
+        }
```
Comment (gandreadis) [inline modules/models/jimple-library-usage-graph/src/test/kotlin/org/cafejojo/schaapi/models/libraryusagegraph/jimple/GeneralizedNodeComparatorValueTest.kt]: Could you perhaps add a test for inequality?

File: modules/models/jimple-library-usage-graph/src/test/kotlin/org/cafejojo/schaapi/models/libraryusagegraph/jimple/JimpleNodeTest.kt [hunk 2]
```diff
@@ -4,11 +4,213 @@ import com.nhaarman.mockito_kotlin.doReturn
...
 import org.jetbrains.spek.api.dsl.describe
 import org.jetbrains.spek.api.dsl.it
 import soot.jimple.DefinitionStmt
 
 internal class JimpleNodeTest : Spek({
+    describe("contained values") {
+        context("non-recursive statements") {
+            it("returns the operator of a throw statement") {
+                val value = mockValue("value")
+                val node = JimpleNode(mockIfStmt(value))
```
Comment (gandreadis) [inline modules/models/jimple-library-usage-graph/src/test/kotlin/org/cafejojo/schaapi/models/libraryusagegraph/jimple/JimpleNodeTest.kt]: mockThrowStmt?

File: modules/models/jimple-library-usage-graph/src/test/kotlin/org/cafejojo/schaapi/models/libraryusagegraph/jimple/JimpleNodeTest.kt [hunk 3]
```diff
@@ -4,11 +4,213 @@ import com.nhaarman.mockito_kotlin.doReturn
...
+
+            it("returns the key of a switch statement") {
+                val value = mockValue("value")
+                val node = JimpleNode(mockSwitchStmt(value))
+
+                assertThat(node.getTopLevelValues()).containsExactly(value)
+                assertThat(node.getValues()).containsExactly(value)
+            }
+
+            it("returns the key of a switch statement") {
```
Comment (gandreadis) [inline modules/models/jimple-library-usage-graph/src/test/kotlin/org/cafejojo/schaapi/models/libraryusagegraph/jimple/JimpleNodeTest.kt]: switch -> invoke?

**Comment by FWDekker** [comment]
@gandreadis Done!

**Pull request merged**